pkg/archive/compress.go

```diff
@@ -22,41 +23,91 @@ const (
 	CompressGzip
 	// CompressXz compression
 	CompressXz
+	// CompressZstd compression
+	CompressZstd
 )
 
 // compressHeaders are used to detect the compression type
 var compressHeaders = map[CompressType][]byte{
 	CompressBzip2: []byte("\x42\x5A\x68"),
 	CompressGzip:  []byte("\x1F\x8B\x08"),
 	CompressXz:    []byte("\xFD\x37\x7A\x58\x5A\x00"),
+	CompressZstd:  []byte("\x28\xB5\x2F\xFD"),
 }
 
 func Compress(r io.Reader, oComp CompressType) (io.Reader, error) {
 	br := bufio.NewReader(r)
-	head, err := br.Peek(10)
+
+	var rComp CompressType
+
+	// Peek length of the magic header for the given compression into the stream
+	head, err := br.Peek(len(compressHeaders[oComp]))
 	if err != nil {
-		return br, err
+		if err == io.EOF {
```

Comment:
I found this bug while writing the tests, any buffer (usually containing uncompressed data) with less than magic header in length end up being `io.EOF` error.